From: Bradley Morgan <include@grrlz.net>
Date: Tue, 28 Jul 2026 20:21:03 +0000
Subject: [PATCH] taskstats: drop the dead NULL attribute check in parse()

Patch series "taskstats: tidy up the cpumask command path".

Two small cleanups from reading kernel/taskstats.c. No functional change in
either one.


This patch (of 2):

taskstats_user_cmd() only calls the cpumask handlers after checking the
same info->attrs[] entry, so parse() never sees a NULL attribute.  Drop
the check and its odd "return 1", which no caller tested for anyway.

No functional change.

Link: https://lore.kernel.org/20260728202104.17839-1-include@grrlz.net
Link: https://lore.kernel.org/20260728202104.17839-2-include@grrlz.net
Signed-off-by: Bradley Morgan <include@grrlz.net>
Cc: Balbir Singh <bsingharora@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---
 kernel/taskstats.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/kernel/taskstats.c b/kernel/taskstats.c
index 74f89b632c0a5..7af47511152d7 100644
--- a/kernel/taskstats.c
+++ b/kernel/taskstats.c
@@ -361,8 +361,6 @@ static int parse(struct nlattr *na, struct cpumask *mask)
 	int len;
 	int ret;
 
-	if (na == NULL)
-		return 1;
 	len = nla_len(na);
 	if (len > TASKSTATS_CPUMASK_MAXLEN)
 		return -E2BIG;
-- 
2.47.3

